src/allotropy/parsers/utils/pandas.py

```diff
@@ -53,50 +53,58 @@ def assert_not_empty_df(df: pd.DataFrame, msg: str) -> pd.DataFrame:
 
 
 T = TypeVar("T", bool, float, int, str)
+# Need to use this instead of type[T] to get mypy to realize primitive can be called to return T
+Type_ = Callable[..., T]
+KeyOrKeys = Iterable[str]
+TypeAndKey = tuple[Type_[T], KeyOrKeys]
+TypeAndKeyAndMsg = tuple[Type_[T], KeyOrKeys, str]
 
 
 class SeriesData:
     def __init__(self, series: pd.Series[Any]) -> None:
         self.series = series
 
-    def __getitem__(
-        self, type_and_key: tuple[Callable[..., T], str | Iterable[str]]
-    ) -> T:
+    def __getitem__(self, type_and_key: TypeAndKey[T] | TypeAndKeyAndMsg[T]) -> T:
```

Comment:
I realized that this would really benefit from a docstring explaining how to use this (what the params are, and that this implements the `series[]` method of get)